rust/src/mask/masking.rs

```diff
@@ -65,21 +73,43 @@ impl Into<MaskObject> for Aggregation {
 
 #[allow(clippy::len_without_is_empty)]
 impl Aggregation {
+    /// Creates a new, empty aggregator for masks or masked models.
     pub fn new(config: MaskConfig) -> Self {
         Self {
             nb_models: 0,
             object: MaskObject::new(config, vec![]),
         }
     }
 
+    /// Gets the length of the aggregated mask or masked model.
     pub fn len(&self) -> usize {
         self.object.data.len()
     }
 
+    /// Gets the mask configuration of the aggregator.
     pub fn config(&self) -> MaskConfig {
         self.object.config
     }
 
+    /// Validates if unmasking of the aggregated, masked model with the given `mask` may be
+    /// safely performed.
+    ///
+    /// This should be checked before calling [`unmask()`], since unmasking may return garbage
+    /// values otherwise.
+    ///
+    /// # Errors
+    /// Fails in one of the following cases:
+    /// - The aggregator has not yet aggregated any models.
+    /// - The number of aggregated masked models is larger than the chosen mask configuration
+    ///   allows.
+    /// - The mask configuration of the aggregator and of the `mask` don't coincide.
+    /// - The length of the aggregated masked model and the `mask` don't coincide.
+    /// - The `mask` itself is invalid.
+    ///
+    /// Even though it does not produce any meaningful values, it may be validated that "unmasking"
```

Comment:
I don't really understand this last sentence

Reply:
the reason lies in the dual use of `MaskObject`. usually we have a masked model and validate that it can be unmasked safely with a mask (both of which are `MaskObject`s of course). but nothing stops a user to apply this validation to a mask itself, so to say "validate if this mask can be unmasked with that mask" or even "validate if this mask can be unmasked with that masked model". this will not produce any meaningful result, but is technically possible due to the chosen types.

Reply:
i'll update the docs to clarify this.